routes/users.js

```diff
@@ -30,8 +30,9 @@ router.get('/:id', async (req, res, next) => {
   }
 });
 router.get('/', async (req, res, next) => {
+  const { limit = 10, offset = 0, username } = req.query;
   try {
-    const users = await userCtrl.getUsers();
+    const users = await userCtrl.getUsers({ pagination: { limit, offset }, filter: { username }});
```

Comment:
don't do this in route, this must be handled in success handler

Reply:
Also make sure you pass total numbers of row to frontend

Reply:
But if I don't pass the pagination options to the controller down to the service , how will the service know the limit and offset for querying? By the way the response structure is handled in `responseHandler` middleware.